src/solvers/dgsem_tree/dg_1d.jl

```diff
@@ -388,6 +436,60 @@ end
     return nothing
 end
 
+@inline function calcflux_fv!(fstar1_L, fstar1_R, u::AbstractArray{<:Any, 3},
+                              mesh::Union{TreeMesh{1}, StructuredMesh{1}},
+                              nonconservative_terms::False,
+                              equations, volume_flux_fv, dg::DGSEM, element, cache,
+                              x_interfaces, reconstruction_mode, slope_limiter)
+    fstar1_L[:, 1] .= zero(eltype(fstar1_L))
+    fstar1_L[:, nnodes(dg) + 1] .= zero(eltype(fstar1_L))
+    fstar1_R[:, 1] .= zero(eltype(fstar1_R))
+    fstar1_R[:, nnodes(dg) + 1] .= zero(eltype(fstar1_R))
+
+    for i in 2:nnodes(dg)
+        #             Reference element:             
+        #  -1 -----------------0----------------- 1 -> x
+        # Gauss-Lobatto-Legendre nodes (schematic for k = 3):
+        #   .         .                 .         .
+        #   ^         ^                 ^         ^
+        # i - 2,    i - 1,              i,      i + 1
+        # mm        ll                  rr      pp
+        # Cell boundaries (schematic for k = 3): 
+        # (note that only the inner three boundaries are stored)
+        #  -1 -----------------0----------------- 1 -> x
+        #   |     |            |             |    |
+        # Cell index:
+        #      1         2              3       4
+
+        # Obtain unlimited values in primal variables
+
+        # Note: If i - 2 = 0 we do not go to neighbor element, as one would do in a finite volume scheme.
+        # Here, we keep it purely cell-local, thus overshoots between elements are not ruled out.
```

Comment:
You can get overshoots because you are doing a completely central reconstruction for the first subcell in

https://github.com/trixi-framework/Trixi.jl/pull/2022/files#diff-791f030db03e61035dbd2e3dd4c70e4ddadeb3fa297231e66e1eb1fdd639d541R58

A "safer" implementation would use a piece-wise constant solution (no reconstruction) for the first and last subcells of the element. Of course, the formal convergence order drops to one in that case. However, in the paper I cite above, we showed (experimentally) that such an implementation is more accurate than the 1st order FV variant.

Would you consider adding the possibility to deactivate the reconstruction procedure at the first and last subcells?

Reply:
Thanks for this valuable information!